### PR TITLE
Add ability to redirect users where they wanted to go

### PR DESCRIPTION
**Pull request by dixpac**
Redirect user to their intended location after logging in.
If nothing is stored in the `session[:user_return_to]` then
redirect to a `root_path`.

Also, `reset_session` is added on login and logout to prevent session
fixation.

Reference -> https://guides.rubyonrails.org/security.html#session-fixation-countermeasures

### CODE REVIEW

File: lib/revise_auth/authentication.rb [hunk 1]
```diff
@@ -51,5 +52,11 @@ def logout
       Current.user = nil
       session.delete(:user_id)
     end
+
+    private
+      # Store user intended url, so we can redirect him there after the login.
+      def store_intended_location
```
Comment (dixpac) [inline lib/revise_auth/authentication.rb:58]: Don't like the method name, but couldn't figure out better name :)

**Comment by drale2k** [comment]
Shouldn't `session[:user_return_to]` be cleared after redirect?

**Comment by excid3** [comment]
> Shouldn't `session[:user_return_to]` be cleared after redirect?

Yeah, probably should do like `session.delete(:user_return_to) || root_path`

**Comment by dixpac** [comment]
Thanks @drale2k , sorry for late response I've missed notification from your comment.

I've solved issue in a different way, added `reset_session` on login and logout to prevent [session fixation](https://guides.rubyonrails.org/security.html#session-fixation-countermeasures), which also solves issue you've found.

File: lib/revise_auth/authentication.rb [hunk 2]
```diff
@@ -52,5 +54,16 @@ def logout
...
+
+    private
+      def redirect_to_login_with_stashed_location
+        stash_intended_location
+        redirect_to login_path, alert: I18n.t("revise_auth.sign_up_or_login")
+      end
+
+      # Store user intended url, so we can redirect him there after the login.
+      def stash_intended_location
+        session[:user_return_to] = request.original_url if request.get?
```
Comment (excid3) [inline lib/revise_auth/authentication.rb:66]: https://github.com/heartcombo/devise/blob/main/lib/devise/controllers/store_location.rb#L56

Wonder if we should strip to just the path as well to be safe?

**Pull request merged**